packages/fx-core/src/component/driver/apiKey/create.ts

```diff
@@ -0,0 +1,264 @@
+        throw appStudioTokenRes.error;
+      }
+      const appStudioToken = appStudioTokenRes.value;
+
+      if (state && state.registrationId) {
+        try {
+          await AppStudioClient.getApiKeyRegistrationById(appStudioToken, state.registrationId);
+          context.logProvider?.info(
+            getLocalizedString(
+              logMessageKeys.skipCreateApiKey,
+              outputEnvVarNames.get(OutputKeys.registrationId)
+            )
+          );
+        } catch (error) {
+          context.logProvider?.warning(
+            getLocalizedString(
+              logMessageKeys.apiKeyNotFound,
+              outputEnvVarNames.get(OutputKeys.registrationId)
+            )
+          );
+        }
+      } else {
+        this.validateArgs(args);
+
+        const apiKey = await this.mapArgsToApiSecretRegistration(context.m365TokenProvider, args);
+        const apiRegistrationRes = await AppStudioClient.createApiKeyRegistration(
+          appStudioToken,
+          apiKey
+        );
+        outputs.set(outputEnvVarNames.get(OutputKeys.registrationId)!, apiRegistrationRes.id!);
+
+        const summary = getLocalizedString(
+          logMessageKeys.successCreateApiKey,
+          apiRegistrationRes.id!
+        );
+        context.logProvider?.info(summary);
+        summaries.push(summary);
+      }
+
+      return {
+        result: ok(outputs),
+        summaries: summaries,
+      };
+    } catch (error) {
+      if (error instanceof UserError || error instanceof SystemError) {
+        context.logProvider?.error(
+          getLocalizedString(logMessageKeys.failedExecuteDriver, actionName, error.displayMessage)
+        );
+        return {
+          result: err(error),
+          summaries: summaries,
+        };
+      }
+
+      const message = JSON.stringify(error);
+      context.logProvider?.error(
+        getLocalizedString(logMessageKeys.failedExecuteDriver, actionName, message)
+      );
+      return {
+        result: err(new UnhandledError(error as Error, actionName)),
+        summaries: summaries,
+      };
+    }
+  }
+
+  // Needs to validate the parameters outside of the function
+  private loadStateFromEnv(
+    outputEnvVarNames: Map<string, string>
+  ): Record<string, string | undefined> {
+    const result: Record<string, string | undefined> = {};
+    for (const [propertyName, envVarName] of outputEnvVarNames) {
+      result[propertyName] = process.env[envVarName];
+    }
+    return result;
+  }
+
+  // Allowed secrets: secret or secret1, secret2
+  // Need to validate secrets outside of the function
+  private parseSecret(apiKeyClientSecret: string): string[] {
+    const secrets = apiKeyClientSecret.trim().split(",");
+    return secrets.map((secret) => secret.trim());
+  }
+
+  private validateSecret(apiKeySecret: string): boolean {
+    if (typeof apiKeySecret !== "string") {
+      return false;
+    }
+
+    const regExp = /^(\w){10,128}(,\s*\w{10,128})*/g;
```

Comment:
why 128?

Reply:
There was a description that client secret should between 10-128 in the design spec, but now it is deleted. Will update this part to max 128.

Reply:
I tried a secret length <10 and will get server error and the error message is not clear, so I prefer to keep the validation here.